### DISCUSSION TITLE
Server can not communicate with client but client to client work

### DISCUSSION THREAD

**Discussion by diktus** [strongswan/strongswan#2137] [General]
Hi,

I have setup strongswan and working well. client can browsing dan also connect each other. But when i try to curl or telnet from server to the client virtual IP it's not working.

The Server is host on AWS EC2. i have stop change source destination check but its seems not working. I need to port forward port 8080 from AWS EC Public IP to one of client. 

Is there anyone have experience to do this scenario?

Thank You

**Comment by tobiasbrunner** [comment]
> But when i try to curl or telnet from server to the client virtual IP it's not working.

What clients? Maybe their firewall or IPsec implementation simply doesn't allow or support this.

**Comment by diktus** [comment]
Hi,

here the firewall configuration. I use UFW on ubuntu server. here the UFW 

*nat
-A POSTROUTING -s 10.10.10.0/24 -o ens5 -m policy --pol ipsec --dir out -j ACCEPT
-A POSTROUTING -s 10.10.10.0/24 -o ens5 -j MASQUERADE
COMMIT

*mangle
-A FORWARD --match policy --pol ipsec --dir in -s 10.10.10.0/24 -o ens5 -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1361:1536 -j TCPMSS --set-mss 1360
COMMIT

-A ufw-before-forward --match policy --pol ipsec --dir in --proto esp -s 10.10.10.0/24 -j ACCEPT
-A ufw-before-forward --match policy --pol ipsec --dir out --proto esp -d 10.10.10.0/24 -j ACCEPT


the service is listen on client with virtual IP 10.10.10.2 on port 8080. another client on ip 10.10.10.1 can curl and telnet but from server it can not communicate

**Comment by TelDragon** [comment]
I also encountered issues with this scenario.

My client has successfully connected to the server and obtained a private IP address. The client can access the internal IP resources of the server normally.

But on the other hand, if I want the server to access the client's intranet, it won't work. Ping doesn't work.
